### ISSUE TITLE
too many "fail to get hist meta for this histogram, possibly a deleted one"

### ISSUE THREAD

**Issue by lance6716** [pingcap/tidb#56740]
## Enhancement
![image](https://github.com/user-attachments/assets/b6ec535d-9cd6-4115-871c-c47123bdcfce)

```
[2024/10/17 23:46:13.170 +08:00] [WARN] [stats_syncload.go:395] ["fail to get hist meta for this histogram, possibly a deleted one"] [table_id=245] [hist_id=35] [is_index=true]
[2024/10/17 23:46:13.180 +08:00] [WARN] [stats_syncload.go:395] ["fail to get hist meta for this histogram, possibly a deleted one"] [table_id=245] [hist_id=35] [is_index=true]
[2024/10/17 23:46:13.183 +08:00] [WARN] [stats_syncload.go:395] ["fail to get hist meta for this histogram, possibly a deleted one"] [table_id=245] [hist_id=35] [is_index=true]
[2024/10/17 23:46:13.218 +08:00] [WARN] [stats_syncload.go:395] ["fail to get hist meta for this histogram, possibly a deleted one"] [table_id=245] [hist_id=35] [is_index=true]
```

**Comment by lance6716** [comment]
use the logger from `SampleLoggerFactory` should be enough

**Comment by hawkingrei** [comment]
may be fixed by #56813
